scripts/install/RegisterItemRelationsService.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\scripts\install;
+
+use oat\oatbox\extension\InstallAction;
+use oat\tao\model\resources\relation\service\ResourceRelationServiceProxy;
+use oat\tao\model\search\SearchProxy;
+use oat\taoAdvancedSearch\model\Resource\Service\ItemRelationsService;
+
+class RegisterItemRelationsService extends InstallAction
+{
+    public function __invoke($params)
+    {
+        if ($this->getServiceManager()->get(SearchProxy::class)->supportCustomIndex()) {
```

Comment:
@bartlomiejmarszal This if is not necessary, if you are using AdvancedSearch you will always be able to add this service to the proxy

Reply:
Add a `FEATURE_FLAG_ADVANCED_SEARCH_DISABLED` check inside the `ItemRelationsService::class` instead.

If you add it here and later switch this FF you will need to manually set this service up. This does not make the app stateless